docs/installation.rst

```diff
@@ -289,6 +299,100 @@ Here's the list of the subpackages and what they enable:
 | winrm               | ``pip install 'apache-airflow[microsoft.winrm]'``   | WinRM hooks and operators                                            |
 +---------------------+-----------------------------------------------------+----------------------------------------------------------------------+
 
+Provider packages
+.................
+
+
+Provider packages context
```

Comment:
My perception could be wrong, but a big benefit I see to this feature is how "dependency-light" it makes the Airflow base. Since all of these providers (which have their own sets of complex dependencies) are no longer included in the base, a user is less likely to run into one-off dependency conflicts when trying to install additional libraries. I wonder if that's worth calling out in the context here? Curious to hear your thoughts @potiuk.

Reply:
Well, it actually was this way already. 

Unless you installed extras, the number of dependencies for the core airlfow was rather small and this has not changed with "provider packages" - we have basically the same set of requirements as before. If you did 'pip install apache-airflow' - the same dependencies are pulled in 1.10 and 2.0. And when you do 'pip install apache-airflow[google]'  - it pulls in the same set of dependencies in 1.10 and 2.0 (but in 2.0 those will be transitive dependencies for google provider, not the extra dependencies). The only difference was that if you have not installed 'google' extra before, the google operators were there in 'airflow.contrib', but they could not be imported because they missed dependencies.

So technically speaking - 2.0 is not any "lighter" in terms of dependencies. It behaves very much the same as it did before.

Maybe we can add some wording around it but literally, the only difference was that previously the classes were there (unusable) and now they are gone if you don't explicitly add 'extra' or install provider package manually. All dependencies remain the same. So for me this is not a differentiating feature. But maybe it is indeed worth spelling out (but I am not sure how :))

Reply:
Ahh that makes sense- I didn't realize this was how it worked pre-2.0! Thanks for clearing it up, probably ok to leave as is for now then.